src/query/Index.cpp

```diff
@@ -91,14 +75,84 @@ std::vector<TriangleMatch> Index::getClosestTriangles(const mesh::Vertex &source
   }
 
   std::vector<TriangleMatch> matches;
-  _pimpl->indices.triangleRTree->query(bgi::nearest(sourceVertex, n),
+  _pimpl->indices.triangleRTree->query(bgi::nearest(sourceCoord, n),
                                        boost::make_function_output_iterator([&](impl::TriangleTraits::IndexType const &match) {
-                                         matches.emplace_back(bg::distance(sourceVertex, _mesh->triangles()[match.second]), match.second);
+                                         matches.emplace_back(bg::distance(sourceCoord, _mesh->triangles()[match.second]), match.second);
                                        }));
   std::sort(matches.begin(), matches.end());
   return matches;
 }
 
+VertexMatch Index::getClosestVertex(const mesh::Vertex &sourceVertex)
+{
+  PRECICE_TRACE();
+  auto matches = getClosestVertices(sourceVertex.getCoords(), 1);
+  // calling .back() on an empty vector is undefined.
+  if (not matches.empty()) {
+    return matches.back();
+  } else {
+    // return a default match with empty index
+    return VertexMatch(-1, std::numeric_limits<double>::max());
+  }
+}
+
+VertexMatch Index::getClosestVertex(const Eigen::VectorXd &sourceCoord)
+{
+  PRECICE_TRACE();
+  auto matches = getClosestVertices(sourceCoord, 1);
+  // calling .back() on an empty vector is undefined.
+  if (not matches.empty()) {
+    return matches.back();
+  } else {
+    // return a default match with empty index
+    return VertexMatch(-1, std::numeric_limits<double>::max());
+  }
+}
+
+EdgeMatch Index::getClosestEdge(const Eigen::VectorXd &sourceCoord)
```

Comment:
The singular version is not required. [YAGNI](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it)